### PR TITLE
CMR-4713 As a Client User, I would like to be able to constrain collection results to collections that have granules or that have been tagged as a CWIC collection

### PR DESCRIPTION
**Pull request by daniel-zamora**
This pull request implements new search feature to include CWIC collections in a has_granules=true search, using has_granules_or_cwic parameter.

### CODE REVIEW

File: search-app/src/cmr/search/services/parameters/conversion.clj [hunk 1]
```diff
@@ -287,6 +288,12 @@
     cqm/match-all
     (qm/->HasGranulesCondition (= "true" value))))
 
+(defmethod common-params/parameter->condition :has-granules-or-cwic
+  [_ _ _ value _]
+  (if (= "unset" value)
```
Comment (chris-durbin) [inline search-app/src/cmr/search/services/parameters/conversion.clj]: "unset" was a holdover from the catalog-rest API for certain parameters. I don't think we need to support it for any new parameters.

File: search-app/src/cmr/search/services/query_execution/has_granules_or_cwic_results_feature.clj [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  "The frequency in seconds of the refresh-has-granules-or-cwic-map-job"
+  ;; default to 1 hour
+  3600)
+
+(def has-granule-cache-key
+  :has-granules-or-cwic-map)
+
+(defconfig cwic-tag
+  "has-granules-or-cwic should also return any collection with configured cwic-tag"
+  {:default "CWIC"})
```
Comment (chris-durbin) [inline search-app/src/cmr/search/services/query_execution/has_granules_or_cwic_results_feature.clj]: We should get the name of the currently used CWIC tag from Mark Reese and use that as the default. I'm guessing it is:
"edsc.portal.cwic.collections"
but we should confirm.

File: search-app/src/cmr/search/services/query_execution/has_granules_or_cwic_results_feature.clj [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                     (fn []
+                       (collection-granule-counts->has-granules-or-cwic-map
+                        (merge
+                         (idx/get-collection-granule-counts context nil)
+                         (get-cwic-collections context nil)))))))
+
+;; This returns a boolean flag with collection results if a collection has any granules in provider holdings
+(defmethod query-execution/post-process-query-result-feature :has-granules-or-cwic
+  [context query elastic-results query-results feature]
+  (assoc query-results :has-granules-or-cwic-map (get-has-granules-or-cwic-map context)))
```
Comment (chris-durbin) [inline search-app/src/cmr/search/services/query_execution/has_granules_or_cwic_results_feature.clj]: Are we returning this field and value in the JSON response? I'm thinking we just need to be able to search by it, not return a field in the response format.

Comment (daniel-zamora) [reply search-app/src/cmr/search/services/query_execution/has_granules_or_cwic_results_feature.clj]: No, I was starting to change all the result handlers then I realized that I probably didn't need to.  Forgot to remove this.

File: system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj [hunk 4]
```diff
@@ -346,3 +350,57 @@
...
+
+    ;; coll 6
+    ;; cwic tagged no granule
+
+    (index/wait-until-indexed)
+    (testing "Search with has-granules-or-cwic feature"
+      (d/refs-match? [coll1 coll3 coll6 coll2]
+                     (search/find-refs :collection
+                                       {:has_granules_or_cwic true}
+                                       {:snake-kebab? false})))
```
Comment (sxu123) [inline system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj]: The Acceptance criteria says "Upon selecting the "Only include collections with granules" checkbox, search results display collections with granules in the CMR and collections with granules that have CWIC tags."  
This doesn't sound right. Could you update it?

Your implementation is to display collections with granules and collections with CWIC tags, which is consistent with the the ticket description.

Comment (daniel-zamora) [reply system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj]: Yea you are right. This needs to change to be more relevant to CMR.

File: search-app/docs/api.md [hunk 5]
```diff
@@ -1612,6 +1612,12 @@ When `has_granules` is set to "true" or "false", results will be restricted to c
 
     curl "%CMR-ENDPOINT%/collections?has_granules=true"
 
+#### <a name="c-has-granules-or-cwic"></a> Find collections with or without granules, or the collection is tagged with the configured CWIC tag.
+
+When `has_granules_or_cwic` can be set to "true" or "false". When true, the results will be restricted to collections with granules or with the configured CWIC tag.  When false, results will be restricted to collections without granules.
```
[the remark this answers is not in the document]
Comment (daniel-zamora) [reply search-app/docs/api.md]: What should it do when passed false? Give an error?

[the remark this answers is not in the document]
Comment (ygliuvt) [reply search-app/docs/api.md]: yeah, we should add validation to only allow "true" value for this param. Then you need to update the tests. Ask @chris-durbin if he thinks this is the way to go too.

**Pull request merged**